backend/src/game.ts

```diff
@@ -0,0 +1,197 @@
+import { Express } from 'express';
+import * as code from './code';
+
+// for clarity, a gameID is just a string
+type UserId = string;
+// used ids for both players and host
+type GameId = string;
+
+// interface for user, mostly blank rn but will keep score or smth later.
+// Userid is stored in the map for now
+export interface User {
+	name: string;
+    answers: number[];
+}
+const usedGame: GameId[] = [];
+
+// define a quiz and question type
+export interface QuizQuestion {
+	questionText: string;
+	answerTexts: string[];
+	correctAnswers: number[];
+	note?: string;
+	time?: number;
+	points?: number;
+}
+
+export interface Quiz {
+	meta: {
+		title: string;
+		author: string;
+		pointDefault: number;
+		timeDefault: number;
+		note?: string;
+	};
+	questions: QuizQuestion[];
+}
+
+export interface Game {
+	users: Map<UserId, User>;
+	hostId: UserId;
+	activeQuestion: number;
+	quizData: Quiz;
+};
+
+function getUsers(game: Game) {
+	return [...(game.users.keys(), game.hostId)];
+}
+
+function beginQuestion() {
+    return;
+}
+
+// first key is gameId
+const games: Map<GameId, Game> = new Map();
+
+export default function registerGameRoutes(app: Express) {
+	app.post('/games', (req, res) => {
+		try {
+            const quizData: Quiz = req.body;
+			const response = {
+				gameId: code.gen(5, []),
+				hostId: code.gen(8, []),
+			};
+			// associate gameId with data and host
+			const data = {
+				users: new Map(),
+				quizData,
+				hostId: response.hostId,
+				activeQuestion: -1,
+			};
+			games.set(response.gameId, data);
+            console.log(response);
+			res.send(response);
+		} catch (e) {
+			// client upload error
+            console.log(e);
+			res.status(400).send('Invalid JSON file');
+			// Todo: further validation
+		}
+	});
+
+	app.post('/games/:gameId/questions/:index/start', (req, res) => {
+		const gameId: GameId = req.params.gameId;
+		const index = parseInt(req.params.index);
+		const game = games.get(gameId);
+
+		// client-requested game error
+		if (game === undefined) {
+			res.status(404).send({ ok: false , err: `Game ${gameId} not found`});
+			return;
+		}
+
+		// we have no way of validating the host yet
+		// client permission 403 error
+
+		const quiz = game.quizData;
+		// out-of-bounds error
+		if (index >= quiz.questions.length) {
+			res.status(404).send({ ok: false , err: `Question ${index} not found`});
+			return;
+		}
+
+		// start accepting answers for the question index
+		game.activeQuestion = index;
+
+		// show question text and answers on both host and player screens
+		beginQuestion();
+        game.users.set('123', { name: 'bob', answers: [] });
```

Comment:
```suggestion
```

Why's this here?

Reply:
Sorry, forgot to take it out while I was testing.